pkg/volume/csi/csi_attacher_test.go

```diff
@@ -387,13 +389,17 @@ func TestAttacherWithCSIDriver(t *testing.T) {
 				if attachID != "" {
 					t.Errorf("Expected empty attachID, got %q", attachID)
 				}
-			}(spec, test.expectVolumeAttachment)
+			}(spec)
 
 			if test.expectVolumeAttachment {
 				expectedAttachID := getAttachmentName("test-vol", test.driver, "fakeNode")
 				status := storage.VolumeAttachmentStatus{
 					Attached: true,
 				}
+				// We want to ensure the second watcher, which is the watcher created in csiAttacher,
+				// has been started before updating the status of attachment.
+				<-attachmentWatchActions
```

Comment:
Can you also add a comment on what is the first watcher?

Reply:
Now we move the `fakeClient.Fake.PrependWatchReactor` out of `newTestWatchPlugin`, there is only one watcher now.

Reply:
The first watcher is the one created in `newTestWatchPlugin`.